src/modules/settings/youtube/Settings.jsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Modal from '../components/Window.jsx';
+import domObserver from '../../../observers/dom.js';
+import DropdownButton from './DropdownButton.jsx';
+import styles from './DropdownButton.module.css';
+
+const BTTV_DROPDOWN_BUTTON_CONTAINER_SELECTOR = 'div[data-a-target="bttv-dropdown-button-container"]';
+const DROPDOWN_MENU_ITEMS_SELECTOR = 'ytd-popup-container #sections #items';
+const ITEMS_SELECTOR = '#items';
+const AVATAR_BUTTON_SELECTOR = '#avatar-btn';
+
+const CHAT_SETTINGS_DROPDOWN_ITEMS_SELECTOR = '#contentWrapper #items';
+const BTTV_CHAT_DROPDOWN_BUTTON_CONTAINER_SELECTOR = 'div[data-a-target="bttv-chat-dropdown-button-container"]';
+
+let handleOpen;
+function setHandleOpen(newHandleOpen) {
+  handleOpen = newHandleOpen;
+}
+
+let mountedPanel;
+let mountedDropdownButton;
+let mountedChatDropdownButton;
+
+let menuItemsListener = null;
+
+export default class SettingsModule {
+  constructor() {
+    this.renderSettings();
+
+    // TODO: re-enable this when we have settings like theatre mode...
+    // domObserver.once(AVATAR_BUTTON_SELECTOR, (node, isConnected) => {
+    //   if (!isConnected) {
+    //     return;
+    //   }
+
+    //   node.addEventListener('click', this.loadDropdownButton);
+    // });
+
+    domObserver.on(CHAT_SETTINGS_DROPDOWN_ITEMS_SELECTOR, (node, isConnected) => {
+      if (!isConnected) {
+        return;
+      }
+
+      this.loadChatAppMenuButton();
+    });
+  }
+
+  renderSettings() {
+    const bttvPanel = document.getElementById('bttvSettingsPanel');
+
+    if (bttvPanel == null) {
+      const panel = document.createElement('div');
+      panel.setAttribute('id', 'bttvSettingsPanel');
+      const chatApp = document.querySelector('#chat');
+
+      if (chatApp == null) {
+        return;
+      }
+
+      chatApp.append(panel);
```

Comment:
why do we append this to chat? shouldn't we append this to body so it doesn't get dropped on a rerender?

Reply:
bump?